src/Equinor.ProCoSys.IPO.WebApi/Synchronization/BusReceiverService.cs

```diff
@@ -35,118 +62,188 @@ public async Task ProcessMessageAsync(PcsTopic pcsTopic, Message message, Cancel
 
             switch (pcsTopic)
             {
+                case PcsTopic.Ipo:
+                    await ProcessIpoEvent(messageJson);
+                    break;
                 case PcsTopic.Project:
-                    var projectEvent = JsonSerializer.Deserialize<ProjectTopic>(messageJson);
-                    _plantSetter.SetPlant(projectEvent.ProjectSchema);
-                    ProcessProjectEvent(projectEvent);
+                    ProcessProjectEvent(messageJson);
                     break;
-
                 case PcsTopic.CommPkg:
-                    var commPkgEvent = JsonSerializer.Deserialize<CommPkgTopic>(messageJson);
-                    _plantSetter.SetPlant(commPkgEvent.ProjectSchema);
-                    ProcessCommPkgEvent(commPkgEvent);
+                    ProcessCommPkgEvent(messageJson);
                     break;
-
                 case PcsTopic.McPkg:
-                    var mcPkgEvent = JsonSerializer.Deserialize<McPkgTopic>(messageJson);
-                    _plantSetter.SetPlant(mcPkgEvent.ProjectSchema);
-                    ProcessMcPkgEvent(mcPkgEvent);
+                    ProcessMcPkgEvent(messageJson);
                     break;
             }
             await _unitOfWork.SaveChangesAsync(token);
         }
 
-        private void ProcessMcPkgEvent(McPkgTopic mcPkgEvent)
+        private void ProcessMcPkgEvent(string messageJson)
+        {
+            var mcPkgEvent = JsonSerializer.Deserialize<McPkgTopic>(messageJson);
+
+            _telemetryClient.TrackEvent(IpoBusReceiverTelemetryEvent,
+                new Dictionary<string, string>
+                {
+                    {BusReceiverTelemetryConstants.Event, IpoTopic.TopicName},
+                    {BusReceiverTelemetryConstants.McPkgNo, mcPkgEvent.McPkgNo},

```

Comment:
possible nullpointer ex on mcPkgEvent. Check on null on line 83 and throw exception?
Same goes for line 105, 121, 139